apps/meteor/ee/app/livechat-enterprise/server/lib/VisitorInactivityMonitor.js

```diff
@@ -90,31 +91,46 @@ export class VisitorInactivityMonitor {
 		const guest = visitor.name || visitor.username;
 		const comment = TAPi18n.__('Omnichannel_On_Hold_due_to_inactivity', { guest, timeout });
 
-		await Promise.all([
+		const result = await Promise.allSettled([
 			LivechatEnterprise.placeRoomOnHold(room, comment, this.user),
 			LivechatRooms.unsetPredictedVisitorAbandonmentByRoomId(room._id),
 		]);
+		if (result.some((r) => r.status === 'rejected')) {
+			logger.error(
+				'Error placing room on hold',
+				result.filter((r) => r.status === 'rejected').map((r) => r.reason),
+			);
+			throw new Error(`Error placing room on hold: ${result.find((r) => r.status === 'rejected').reason}`);
+		}
 	}
 
 	async handleAbandonedRooms() {
 		const action = settings.get('Livechat_abandoned_rooms_action');
 		if (!action || action === 'none') {
 			return;
 		}
-		await Promise.all(
-			LivechatRooms.findAbandonedOpenRooms(new Date()).forEach((room) => {
+		const result = await Promise.allSettled(
+			LivechatRooms.findAbandonedOpenRooms(new Date()).map(async (room) => {
 				switch (action) {
 					case 'close': {
-						this.closeRooms(room);
+						await this.closeRooms(room);
 						break;
 					}
 					case 'on-hold': {
-						Promise.await(this.placeRoomOnHold(room));
+						await this.placeRoomOnHold(room);
 						break;
 					}
 				}
 			}),
 		);
+		const failed = result.filter((r) => r.status === 'rejected');
+		if (failed.length) {
+			logger.error(
+				'Failed to handle abandoned omnichannel rooms',
+				failed.map((r) => r.reason),
```

Comment:
Will this print something good or something with a weird format? Should we join them? Or stringify?

Reply:
It kind of depends on the error being thrown within the promise. So if it's a normal error like for eg: `throw new Error('Some error msg')`), then the `reason` prop would only have the error string, i.e. `Some error msg`. However, if it's a custom error wrapper (like `Meteor.Error`), then it will have its own object.

In any case, I think the logger which we're using should be able to parse then so don't think we'd explicitly need to stringify them or process them in any special way